atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/CacheStoreImpl.java

```diff
@@ -98,6 +96,18 @@ public TransactionScopedCache getReadOnlyCache(StartTimestamp timestamp) {
                 .orElseGet(() -> NoOpTransactionScopedCache.create().createReadOnlyCache(CommitUpdate.invalidateAll()));
     }
 
+    private void validateStateSize() {
+        if (cacheMap.size() > maxCacheCount) {
+            log.warn(
+                    "Transaction cache store has exceeded maximum concurrent caches. This likely indicates a memory"
+                            + " leak",
+                    SafeArg.of("cacheMapSize", cacheMap.size()),
```

Comment:
Let's also log the lowest timestamp in the map? This should let us figure out how old the oldest entry is

Reply:
It's not a navigable map in any way, so that's hard to do :(